source/common/upstream/load_balancer_impl.h

```diff
@@ -787,6 +787,9 @@ class LoadBalancerSubsetInfoImpl : public LoadBalancerSubsetInfo {
       }
     }
   }
+  LoadBalancerSubsetInfoImpl()
```

Comment:
Yes, Singleton requires a default c'tor. I personally find this straightfoward and readable. It's also consistent with my previous changes, eg #24243

Reply:
I agree that it is straightforward and readable, just not sure if it is intended (for example, this can now be used to create a vector of `LoadBalancerSubsetInfoImpl`).
I don't feel strongly about this, just wanted to highlight that making `ConstSingleton<>` probably better support the intended use-case.